src/main/resources/cerner-search/searchById.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~  Copyright (c) 2018, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+~
+~  WSO2 Inc. licenses this file to you under the Apache License,
+~  Version 2.0 (the "License"); you may not use this file except
+~  in compliance with the License.
+~  You may obtain a copy of the License at
+~
+~   http://www.apache.org/licenses/LICENSE-2.0
+~
+~  Unless required by applicable law or agreed to in writing,
+~  software distributed under the License is distributed on an
+~  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+~  KIND, either express or implied.  See the License for the
+~  specific language governing permissions and limitations
+~  under the License.
+-->
+<template name="searchById" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="type" description="The name of a resource type"/>
+    <parameter name="id" description="The logical Id of the resource"/>
+    <sequence>
+        <property name="uri.var.type" expression="$func:type"/>
+        <property name="uri.var.id" expression="$func:id"/>
+        <script language="js">
+            <![CDATA[
+            var id = mc.getProperty('uri.var.id');
+            urlQuery = '';
+            if  (id != null && id != "") {
+                urlQuery += '_id=' + id + '&';
+            }
+            if  (urlQuery != null && urlQuery  != "") {
+                urlQuery = '?' + urlQuery.toString().substring(0, urlQuery.toString().lastIndexOf("&"));
+            }
+            mc.setProperty('uri.var.urlQuery',urlQuery);
+            ]]>
+        </script>
+        <filter source="boolean($ctx:uri.var.id)" regex="false">
+            <then>
+                <log category="ERROR" description="FAULT" level="custom">
+                    <property name="ERROR" value="No supported Mandatory search parameters provided"/>
+                </log>
+                <payloadFactory media-type="json">
+                    <format>{ "Error":{ "errorType":"InvalidParameter","details":"at least the id parameter must be
+                        provided " } }
+                    </format>
+                    <args/>
+                </payloadFactory>
+                <header name="To" scope="default" action="remove"/>
+                <property name="RESPONSE" value="true" scope="default" type="STRING"/>
+                <property name="HTTP_SC" value="400" scope="axis2" type="STRING"/>
+                <send/>
+            </then>
+            <else>
+                <header name="Accept" scope="transport" value="application/json+fhir"/>
+                <call>
+                    <log category="INFO" level="full" separator=","/>
+                    <endpoint>
+                        <http method="Get"
+                              uri-template="{uri.var.base}/{uri.var.type}{+uri.var.urlQuery}"/>
```

Comment:
Is there any change to have special characters in the _id? If not, we can remove the reserved expansion (+) in the URI template.